### PR TITLE
Deprecate `--flag: bool` in custom command

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
While #11057 is merged, it's hard to tell the difference between `--flag: bool` and `--flag`, and it makes user hard to read custom commands' signature, and hard to use them correctly.

After discussion, I think we can deprecate `--flag: bool` usage, and encourage using `--flag` instead.

# User-Facing Changes
The following code will raise warning message, but don't stop from running.
```nushell
❯ def florb [--dry-run: bool, --another-flag] { "aaa" };  florb
Error:   × Deprecated: --flag: bool
   ╭─[entry #7:1:1]
 1 │ def florb [--dry-run: bool, --another-flag] { "aaa" };  florb
   ·                       ──┬─
   ·                         ╰── `--flag: bool` is deprecated. Please use `--flag` instead, more info: https://www.nushell.sh/book/custom_commands.html
   ╰────

aaa
```

cc @kubouch 

# Tests + Formatting
Done

# After Submitting
- [ ] Add more information under https://www.nushell.sh/book/custom_commands.html to indicate `--dry-run: bool` is not allowed, 
- [ ] remove `: bool` from custom commands between 0.89 and 0.90

### CODE REVIEW

**Comment by amtoine** [comment]
i don't think this is a breaking change as both behaviour are still allowed :wink:

File: crates/nu-protocol/src/parse_warning.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Error, Diagnostic, Serialize, Deserialize)]
+pub enum ParseWarning {
+    #[error("Deprecated: {0}")]
+    DeprecatedWarning(
+        String,
+        String,
+        #[label = "`{0}` is deprecated. Please use `{1}` instead, more info: https://www.nushell.sh/book/custom_commands.html"]
```
Comment (amtoine) [inline crates/nu-protocol/src/parse_warning.rs:12]: i would add the release from which the feature should be removed here, maybe 0.90?

**Review by amtoine — COMMENTED** [review]
the changes look great to me :ok_hand: 

most of them will disappear anyway once the annotation gets removed.

File: crates/nu-protocol/src/parse_warning.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-protocol/src/parse_warning.rs:12]: Yeah I think so.  I can remove it on 0.90

Comment (amtoine) [reply crates/nu-protocol/src/parse_warning.rs:12]: nice :ok_hand: 

actually, what i meant was the following change, making it explicit that the annotation support will be removed in a given version :wink: 
```suggestion
        #[label = "`{0}` is deprecated and will be removed in 0.90. Please use `{1}` instead, more info: https://www.nushell.sh/book/custom_commands.html"]
```

**Review by amtoine — APPROVED** [review]
looks good, thanks @WindSoilder :pray:

**Pull request merged**